### DISCUSSION TITLE
Add support for loading FamiStudio tracker music (`.nsf`)

### DISCUSSION THREAD

**Discussion by skyebleuu** [godotengine/godot-proposals#11263] [Engine Core]
### Describe the project you are working on

retro styled rpg based around 2 worlds.

### Describe the problem or limitation you are having in your project

the person making music for the game can export a way smaller file than an ogg/mp3/whatever and we cant import it into godot.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

people using famistudio to make music could import nsf files to use their music more easily, quickly, and can save space for the final result of the pck file or just the exported version in general.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

it could be a stream type or some sort of conversion when importing, and could work like a wav, mp3, or ogg file easily.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

there isnt really a way to add nsf support without modifying the source for godot.

### Is there a reason why this should be core and not an add-on in the asset library?

its good for qol when importing music from famistudio and convenient for storage.

**Comment by aaronfranke** [comment]
> there isnt really a way to add nsf support without modifying the source for godot.

That's not true. You can implement support for your own custom format in GDScript by extending `EditorImportPlugin`: https://docs.godotengine.org/en/stable/classes/class_editorimportplugin.html

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

___

- Related to https://github.com/godotengine/godot-proposals/issues/10077.

If tracker music is ever to be supported in core, it should use a standard format, not a format specific to a specific tracker software or DAW.
